### PR TITLE
Refresh cpu manager state if inconsistant

### LINKED ISSUE #79823: Kubelet cpu manager should refresh cpuset if cpu topology changed
**What happened**:
A node was previously installed with kubelet and use the static manager
as the node only contained 1 core, the /var/lib/kubelet/cpu_manager_state was initialized with
```json
{"policyName":"static","defaultCpuSet":"0","checksum":885245905}
```
later, the node was upgrade to 4 cores, and the file was not refreshed and the newly added cpus cannot be used by any workload anymore

**What you expected to happen**:
the kubelet should be aware of this cpu topology change ,and added the new cores to the manager state file.

**How to reproduce it (as minimally and precisely as possible)**:
install kubelet with cpu manage enabled. then upgrade the VM with more cores
the cpu manager state is not refreshed

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.12.2
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by peter-wangxu**
The commit tries to solve following 2 cases:
1. The machine has been scaled up, the default cpuset
   should be updated accordingly so that container can use it.
2. The machine has been scaled down, the default cpuset
   should be updated accordingly so that invalid cpuset was
   removed which avoid container startup due to out-of-range cpuset
   for container runtime

In both cases, each container should be added via AddContainer for
potential new cpuset.

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #79823

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by mattjmcnaughton — COMMENTED** [review]
/unassign
I don't have a ton of context on this part of the code base.

/cc @ConnorDoyle 
Connor, mind taking a look? I think you've worked on this part of the code base in the past :)

File: pkg/kubelet/cm/cpumanager/policy_static_test.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func TestStaticPolicyStart(t *testing.T) {
 			description: "non-corrupted state",
 			topo:        topoDualSocketHT,
 			stAssignments: state.ContainerCPUAssignments{
-				"0": cpuset.NewCPUSet(0),
+				"conainer0": cpuset.NewCPUSet(0),
```
Comment (matthyx) [inline pkg/kubelet/cm/cpumanager/policy_static_test.go]: nit: probably `container0` here

**Comment by matthyx** [comment]
@peter-wangxu please fix your test and linting issue

**Comment by matthyx** [comment]
Strange this error... are the tests passing locally?

**Comment by peter-wangxu** [comment]
> Strange this error... are the tests passing locally?
@matthyx 
I just rebased and fixed some newly added tests after my initial version. let's see the result

**Pull request closed without merging**